### DISCUSSION TITLE
Splitting minner and market process when error

### DISCUSSION THREAD

**Discussion by PerrywLiu** [filecoin-project/lotus#7636] [Storage Provider]
ERROR: checking sealer API: could not get DialArgs: parse "file:/go/src/github.com/filecoinproject/lotus-1.11.2-rc2/chain/stmgr/utils.go,line:47f github.com/filecoin-project/lotus/chain/stmgr.init.0\nMINER_API_INFO=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.peHxpV_hLMoqymScRPLacDQiBkB-fR5ENhaXtj4Hspw:/ip4/127.0.0.1/tcp/2345/http": net/url: invalid control character in URL

when I start lotusmarkets, up error happens, I compare the url format with lotus daemon, it alomest same. How I can resolev this problem?

**Comment by PerrywLiu** [comment]
1. 

> ERROR: checking sealer API: could not get DialArgs: parse "file:/Users/liupw/go/src/github.com/filecoinproject/lotus-1.11.2-rc2/chain/stmgr/utils.go,line:47f github.com/filecoin-project/lotus/chain/stmgr.init.0\nMINER_API_INFO=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.peHxpV_hLMoqymScRPLacDQiBkB-fR5ENhaXtj4Hspw:/ip4/127.0.0.1/tcp/2345/http": net/url: invalid control character in URL

**Reply by stuberman** [reply]
Please check your token address, make sure the response is exactly the same, I wonder about the '-' and the '_'

Run this to get your token and then paste the actual string use use in the env or config.toml

`lotus-miner auth api-info --perm admin`

**Reply by stuberman** [reply]
Do not paste your entire API token here, just ---- out most of it, as this can be used to hack your machine.

**Reply by stuberman** [reply]
Here are examples I use
**env**
export MINER_API_INFO=eynnn.nnnn_nnnnn-nnnn:/ip4/192.168.1.95/tcp/2345/http

**config.toml in .lotusmarkets**
[Subsystems]
  EnableMining = false
  EnableSealing = false
  EnableSectorStorage = false
  EnableMarkets = true
  SealerApiInfo = "[token]:/ip4/192.168.1.95/tcp/2345/http"
  SectorIndexApiInfo = "[token]:/ip4/192.168.1.95/tcp/2345/http"

**Reply by PerrywLiu** [reply]
> Do not paste your entire API token here, just ---- out most of it, as this can be used to hack your machine.

Thanks for your reminding, I use the 2k localnet test, so I ignore the security problem.

**Reply by PerrywLiu** [reply]
I check my env and config.toml， and change it as you provider method, but it still reponse the same error. Maybe I make some error in somewhere. Thank you verymuch, let me check other env.

**Reply by Angelo-gh3990** [reply]
Did you fill in the exact IP address on the miner side because it reads it from the miner API:

```
[API]
  ListenAddress = "/ip4/x.x.x.x/tcp/2345/http"
  RemoteListenAddress = "x.x.x.x:2345"
  Timeout = "30s"
```

**Reply by PerrywLiu** [reply]
yes, I do it ,and restart the proving/mining miner, has the same error like this:

`Validated 51 messages (1 per second)

Done!
2021-11-18T11:29:15.724+0800	INFO	main	lotus-miner/init_restore.go:145	Checking if repo exists
2021-11-18T11:29:15.724+0800	INFO	main	lotus-miner/init_restore.go:160	Initializing repo
2021-11-18T11:29:15.724+0800	INFO	repo	repo/fsrepo.go:140	Initializing repo at '/Users/liupw/.lotusmarketsDevnet'
2021-11-18T11:29:15.725+0800	DEBUG	lock	go-fs-lock@v0.0.6/fslock.go:62	Checking lock
2021-11-18T11:29:15.726+0800	DEBUG	lock	go-fs-lock@v0.0.6/fslock.go:64	File doesn't exist: /Users/liupw/.lotusmarketsDevnet/repo.lock
2021-11-18T11:29:15.726+0800	INFO	main	lotus-miner/init_restore.go:173	Restoring config
ERROR: checking sealer API: could not get DialArgs: parse "file:/Users/liupw/go/src/github.com/filecoinproject/lotus-1.11.2-rc2/chain/stmgr/utils.go,line:47f github.com/filecoin-project/lotus/chain/stmgr.init.0\nMINER_API_INFO=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.peHxpV_hLMoqymScRPLacDQiBkB-fR5ENhaXtj4Hspw:/ip4/127.0.0.1/tcp/2345/http": net/url: invalid control character in URL`

and my ./lotusminer/config.toml file as this:

`[API]
  
  ListenAddress = "/ip4/127.0.0.1/tcp/2345/http"

  RemoteListenAddress = "127.0.0.1:2345"

  Timeout = "30s"

` 

the market/confg.toml as this :

`[API]

  ListenAddress = "/ip4/127.0.0.1/tcp/8787/http"

  RemoteListenAddress = "127.0.0.1:8787"

  Timeout = "30s"



[Libp2p]
 
  AnnounceAddresses = ["/ip4/127.0.0.1/tcp/24001"]


[Subsystems]

  EnableMining = false

  EnableSealing = false

  EnableSectorStorage = false

  EnableMarkets = true

  SealerApiInfo = "MINER_API_INFO=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.peHxpV_hLMoqymScRPLacDQiBkB-fR5ENhaXtj4Hspw:/ip4/127.0.0.1/tcp/2345/http"

  SectorIndexApiInfo = "MINER_API_INFO=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.peHxpV_hLMoqymScRPLacDQiBkB-fR5ENhaXtj4Hspw:/ip4/127.0.0.1/tcp/2345/http"


[Dealmaking]
  ConsiderOnlineStorageDeals = true
  ConsiderOfflineStorageDeals = true

  ConsiderOnlineRetrievalDeals = true

  ConsiderOfflineRetrievalDeals = true

  ConsiderVerifiedStorageDeals = true

  ConsiderUnverifiedStorageDeals = true

  PieceCidBlocklist = []

  ExpectedSealDuration = "24h0m0s"

  PublishMsgPeriod = "1h0m0s"

  MaxDealsPerPublishMsg = 8

  Filter = "/absolute/path/to/storage_filter_program"
  RetrievalFilter = "/absolute/path/to/retrieval_filter_program"

  [Dealmaking.RetrievalPricing]


    [Dealmaking.RetrievalPricing.Default]


    [Dealmaking.RetrievalPricing.External]




`

**Comment by LexLuthr** [comment]
@PerrywLiu Were you able to fix this error? If yes, can you please let us know what was the solution?

**Reply by LexLuthr** [reply]
I am archiving this discussion as there has been no reply for more than 7 days.
